src/coreclr/src/vm/threads.cpp

```diff
@@ -54,6 +54,41 @@
 #include "eventpipebuffermanager.h"
 #endif // FEATURE_PERFTRACING
 
+static PortableTailCallFrame g_sentinelTailCallFrame = { NULL, NULL, NULL };
+
+TailCallTls::TailCallTls()
+    : m_frame(&g_sentinelTailCallFrame)
+    , m_argBuffer(NULL)
+    , m_argBufferSize(0)
+    , m_argBufferGCDesc(NULL)
+{
+}
+
+void* TailCallTls::AllocArgBuffer(size_t size, void* gcDesc)
+{
+    m_argBufferSize = size;
```

Comment:
Assert that `m_argBuffer` is NULL? We should not be overwriting any existing block.

Reply:
It will also need to be zeroed in `FreeArgBuffer` below then. Is that preferable?

Reply:
Yes, it is worth it - avoiding dangling pointers helps with diagnosability.